lib/parse.js

```diff
@@ -22,15 +22,15 @@ export function transformPayloadForOpenAICompatibility(payload) {
 
 /** @type {import('..').VerifyAndParseRequestInterface} */
 export async function verifyAndParseRequest(body, signature, keyID, options) {
-  const isValidRequest = await verifyRequestByKeyId(
+  const { isValid } = await verifyRequestByKeyId(
     body,
     signature,
     keyID,
     options
   );
 
   return {
-    isValidRequest,
+    isValidRequest: isValid,
```

Comment:
The `verifyAndParseRequest` method should also accept `options.cache` and return a `cache` key. It's the most high-level of the methods that probably most users end up using right now. Without exposing these, they won't be able to implement key caching.

Reply:
Thanks @gr2m, I'll get to work on this 👁️

Reply:
> The verifyAndParseRequest method should also accept options.cache and return a cache key. It's the most high-level of the methods that probably most users end up using right now. Without exposing these, they won't be able to implement key caching.

@gr2m - https://github.com/copilot-extensions/preview-sdk.js/pull/81/commits/81a6f3387ba94444b936223123f39eafa23835c2 is my interpretation of this --when you have cycles, may I please ask for your review in that's what you envisioned? Happy to correct anything that I may have missed or errantly understood! 🙇

Reply:
looks good to me now, thank you 👍🏼